tripy/tripy/backend/mlir/memref.py

```diff
@@ -37,11 +37,27 @@ def create_empty_memref(shape, dtype, device=tp_device("gpu"), stream=None):
     )
 
 
+def create_empty_memref(
+    shape: Sequence[int],
+    dtype: str,
+    device: tp_device = tp_device("gpu"),
+    stream=None,
+    use_cache: bool = True,
+):
+    """
+    Creates an empty memref, used for allocating memory.
+    Caches the result for subsequent calls with the same parameters.
```

Comment:
I am not sure if I understand caching of `memref`s. How can we share `memref` across different tensors with same shapes, dtypes?

Reply:
We should only cache empty memrefs then right? Should we assert that in `_cached_create_memref`? Or maybe just make it internal to this function?

Reply:
Added an assert and made the function internal.